src/client/Microsoft.Identity.Client/Internal/Logger/MsalLoggerExtensions.cs

```diff
@@ -0,0 +1,156 @@
+
+using System;
+using System.Diagnostics.Tracing;
+using System.IO;
+using System.Runtime.CompilerServices;
+using Microsoft.Identity.Client.Internal.Logger;
+using Microsoft.IdentityModel.Logging.Abstractions;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Extension methods for ICoreLogger
+    /// </summary>
+    internal static class MsalLoggerExtensions
+    {
+        public static void Always(this IIdentityLogger logger, string message)
+        {
+            var logEntry = new LogEntry();
+            logEntry.Message = message;
+            logEntry.EventLevel = EventLevel.LogAlways;
+
+            logger.Log(logEntry);
+        }
+
+        public static void AlwaysPii(this IIdentityLogger logger, string messageWithPii, string messageScrubbed)
+        {
+            var logEntry = new LogEntry();
+            logEntry.Message = logger.IsPiiEnabled ? messageWithPii: messageScrubbed;
+            logEntry.EventLevel = EventLevel.LogAlways;
+
+            logger.LogToAvailableLogger(logEntry, true);
+        }
+
+        public static void Error(this IIdentityLogger logger, string message)
+        {
+            var logEntry = new LogEntry();
+            logEntry.Message = message;
+            logEntry.EventLevel = EventLevel.Error;
+
+            logger.Log(logEntry);
+        }
+
+        public static void Warning(this IIdentityLogger logger, string message)
+        {
+            var logEntry = new LogEntry();
+            logEntry.Message = message;
+            logEntry.EventLevel = EventLevel.Warning;
+
+            logger.Log(logEntry);
+        }
+
+        public static void Info(this IIdentityLogger logger, string message)
+        {
+            var logEntry = new LogEntry();
+            logEntry.Message = message;
+            logEntry.EventLevel = EventLevel.Informational;
+
+            logger.Log(logEntry);
+        }
+
+        public static void InfoPii(this IIdentityLogger logger, string messageWithPii, string messageScrubbed)
+        {
+            var logEntry = new LogEntry();
+            logEntry.Message = logger.IsPiiEnabled ? messageWithPii : messageScrubbed;
+            logEntry.EventLevel = EventLevel.Informational;
+
+            logger.LogToAvailableLogger(logEntry, true);
+        }
+
+        public static void Verbose(this IIdentityLogger logger, string message)
+        {
+            var logEntry = new LogEntry();
+            logEntry.Message = message;
+            logEntry.EventLevel = EventLevel.Verbose;
+
+            logger.Log(logEntry);
+        }
+
+        //Since there is not a good way to determine if the pii value is needed by the logger at the moment, we need to make a check.
+        private static void LogToAvailableLogger(this IIdentityLogger logger, LogEntry logEntry, bool containsPii)
+        {
+            //Not the best way to do this. Needs refactoring.
+            if (containsPii && logger is LegacyIdentityLoggerAdapter loggerAdapter)
```

Comment:
This is clearly not the best way to do this but it does demonstrate that we need some way of sending the containsPii value to the legacy logger callback.

Reply:
The log adapter should encapsulate this better.